examples/cloud-datafusion-functions-plugins/README.md

```diff
@@ -0,0 +1,371 @@
+- [Reusable Plugins](#reusable-plugins-for-cloud-data-fusion-cdf--cdap)
+    - [Overview](#overview)
+    - [CheckPointRead Action, CheckPointUpdate Action](#checkpointread-action-checkpointupdate-action)
+        - [Dependencies](#dependencies)
+            - [Setting up Firestore](#setting-up-firestore)
+            - [Set Runtime Arguments](#set-runtime-arguments)
+    - [CopyTable Action](#copytable-action)
+    - [DropTable Action](#droptable-action)
+    - [TruncateTable Action](#truncatetable-action)
+- [Putting it all together into a Pipeline](#putting-it-all-together-into-a-pipeline)
+  - [CheckPointReadAction](#checkpointreadaction)
+  - [TruncateTableAction](#truncatetableaction)
+  - [Database source](#database-source)
+  - [BigQuery sink](#bigquery-sink)
+  - [MergeLastUpdateTSAction](#mergelastupdatetsaction)
+  - [CheckPointUpdateAction](#checkpointupdateaction)
+- [Building the CDF/CDAP Plugin .jar file and deploying into CDF/CDAP](#building-the-cdfcdap-plugin-jar-file-and-deploying-into-cdfcdap)
+
+# Reusable Plugins for Cloud Data Fusion (CDF) / CDAP
+
+## Overview
+The CDF/CDAP plugins detailed below can be reused in the context of data pipelines.
+
+Let's say you run your incremental pipeline once every 5 minutes. When running an incremental pipeline, you have to filter the records by a specific field (e.g., lastUpdateDateTime of records > latest watermark value - buffer time) so it will sync the records that were updated since your last incremental sync. Subsequently, a merge and dedupe step is done to make sure only new/updated are synced into the destination table.
```

Comment:
```suggestion
Let's say you run your incremental pipeline once every 5 minutes. When running an incremental pipeline, you have to filter the records by a specific field (e.g., `lastUpdateDateTime` of records > latest watermark value - buffer time) so it will sync the records that were updated since your last incremental sync. Subsequently, a merge and dedupe step is done to make sure only new/updated are synced into the destination table.
```
Please put code-related text in code font. See: https://developers.google.com/style/code-in-text

Reply:
thank you @mik-laj